packages/atomic/src/components/atomic-smart-snippet/atomic-smart-snippet-source.tsx

```diff
@@ -0,0 +1,104 @@
+  @Prop() smartSnippet: SmartSnippet | undefined;
+  @Prop() isSuggestion!: boolean;
+  public error!: Error;
+
+  @Listen('atomic/resolveResult')
+  public resolveResult(event: ResultContextEvent<Result>) {
+    event.preventDefault();
+    event.stopPropagation();
+    if (this.source) event.detail(this.source);
+  }
+
+  render() {
+    const interactiveResult = buildInteractiveResult(this.bindings.engine!, {
+      options: {result: this.source},
+    });
+
+    if (this.smartSnippet && !this.isSuggestion) {
+      return (
+        <section aria-label={this.bindings.i18n.t('smart-snippet-source')}>
+          <div part="source-url">
+            <LinkWithResultAnalytics
+              title={this.source.clickUri}
+              href={this.source.clickUri}
+              target="_self"
+              onSelect={() => this.smartSnippet!.selectSource()}
+              onBeginDelayedSelect={() =>
+                this.smartSnippet!.beginDelayedSelectSource()
+              }
+              onCancelPendingSelect={() =>
+                this.smartSnippet!.cancelPendingSelectSource()
+              }
+            >
+              {this.source.clickUri}
+            </LinkWithResultAnalytics>
+          </div>
+          <div part="source-title" class="mb-6">
+            <LinkWithResultAnalytics
+              title={this.source.title}
+              href={this.source.clickUri}
+              target="_self"
+              onSelect={() => this.smartSnippet!.selectSource()}
+              onBeginDelayedSelect={() =>
+                this.smartSnippet!.beginDelayedSelectSource()
+              }
+              onCancelPendingSelect={() =>
+                this.smartSnippet!.cancelPendingSelectSource()
+              }
+            >
+              <atomic-result-text
+                field="title"
+                default="no-title"
+              ></atomic-result-text>
+            </LinkWithResultAnalytics>
+          </div>
+        </section>
+      );
+    }
+
+    return [
+      <LinkWithResultAnalytics
+        title={this.source.clickUri}
+        href={this.source.clickUri}
+        target="_self"
+        part="source-url"
+        onSelect={() => interactiveResult.select()}
+        onBeginDelayedSelect={() => interactiveResult.beginDelayedSelect()}
+        onCancelPendingSelect={() => interactiveResult.cancelPendingSelect()}
+      >
+        {this.source.clickUri}
+      </LinkWithResultAnalytics>,
+      <LinkWithResultAnalytics
+        title={this.source.title}
+        href={this.source.clickUri}
+        target="_self"
+        part="source-title"
+        onSelect={() => interactiveResult.select()}
+        onBeginDelayedSelect={() => interactiveResult.beginDelayedSelect()}
+        onCancelPendingSelect={() => interactiveResult.cancelPendingSelect()}
+      >
+        <atomic-result-text
+          field="title"
+          default="no-title"
+        ></atomic-result-text>
+      </LinkWithResultAnalytics>,
+    ];
```

Comment:
Yeah... now looking at both I can see they're a bit different... Using a separate component for the `atomic-smart-snippet.tsx` might be overkill, just handling the event directly like it was should be enough.

That means we would only use this new component for `atomic-smart-snippet-suggestions.tsx`

Reply:
actually, I think we should do the opposite: only use it for `smart-snippet`. The problem here is that smart snippet suggestions don't support `select`/`beginDelayedSelect`/`cancelPendingSelect` yet, which they will in the future.

When I add the missing functions to smart snippet suggestions, I will replace the source in smart snippet suggestions to use this component instead.